src/Integration.Vsix/CFamily/CFamilyReproducerCommand.cs

```diff
@@ -0,0 +1,166 @@
+        private readonly ISonarLanguageRecognizer sonarLanguageRecognizer;
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Initializes the singleton instance of the command.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        public static async Task InitializeAsync(AsyncPackage package, ILogger logger)
+        {
+            // Switch to the main thread - the call to AddCommand in Command1's constructor requires
+            // the UI thread.
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(package.DisposalToken);
+
+            var monitorSelection = await package.GetServiceAsync(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
+            var adapterService = await package.GetMefServiceAsync<IVsEditorAdaptersFactoryService>();
+            var docLocator = new ActiveDocumentLocator(monitorSelection, adapterService);
+
+            var languageRecognizer = await package.GetMefServiceAsync<ISonarLanguageRecognizer>();
+
+            IMenuCommandService commandService = (IMenuCommandService)await package.GetServiceAsync(typeof(IMenuCommandService));
+            Instance = new CFamilyReproducerCommand(commandService, docLocator, languageRecognizer, logger);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DisableRuleCommand"/> class.
+        /// Adds our command handlers for menu (commands must exist in the command table file)
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        /// <param name="menuCommandService">Command service to add command to, not null.</param>
+        internal /* for testing */ CFamilyReproducerCommand(IMenuCommandService menuCommandService,
+            IActiveDocumentLocator activeDocumentLocator, ISonarLanguageRecognizer languageRecognizer, ILogger logger)
+        {
+            if (menuCommandService == null)
+            {
+                throw new ArgumentNullException(nameof(menuCommandService));
+            }
+
+            this.activeDocumentLocator = activeDocumentLocator ?? throw new ArgumentNullException(nameof(activeDocumentLocator));
+            this.sonarLanguageRecognizer = languageRecognizer ?? throw new ArgumentNullException(nameof(languageRecognizer));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            var menuCommandID = new CommandID(CommandSet, CommandId);
+            menuItem = new OleMenuCommand(Execute, null, QueryStatus, menuCommandID);
+            menuCommandService.AddCommand(menuItem);
+        }
+
+        private void QueryStatus(object sender, EventArgs args)
+        {
+            // Settings enabled to false will stop the command being executed.
+            // VS will print "The command is not available" in the Command window.
+            try
+            {
+                menuItem.Visible = false;
```

Comment:
The command is only accessible from the command window so it never needs to be visible (although since it isn't attached to a menu it doesn't make a lot of difference).